config/src/main/java/com/typesafe/config/impl/Tokenizer.java

```diff
@@ -478,13 +515,14 @@ private Token pullQuotedString() throws ProblemException {
             if (sb.length() == 0) {
                 int third = nextCharRaw();
                 if (third == '"') {
-                    appendTripleQuotedString(sb);
+                    sbOrig.appendCodePoint(third);
+                    appendTripleQuotedString(sb, sbOrig);
                 } else {
                     putBack(third);
                 }
-            }
 
-            return Tokens.newString(lineOrigin, sb.toString());
+            }
+            return Tokens.newString(lineOrigin, sb.toString(), sbOrig.toString());
```

Comment:
it might be worthwhile to jump through some hoops to try to use the same string twice here, in the common case with no escape sequences, so we don't end up with double the memory. I think that would mean: 1) record triple vs. double quotes flag separately, instead of putting the quotes in the sbOrig string, and 2) check here whether sb.toString.equals(sbOrig.toString).  ?

Now that I say that, I'm fine with leaving it out of this PR though. we can optimize it later.